addon/components/frost-gauge.js

```diff
@@ -0,0 +1,215 @@
+import Ember from 'ember'
+const {isPresent} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
+import layout from '../templates/components/frost-gauge'
+
+const directions = {
+  CLOCKWISE: 1,
+  COUNTERCLOCKWISE: -1
+}
+
+export default Component.extend({
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+
+  attributeBindings: ['style'],
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {
+    // required
+    data: PropTypes.arrayOf(PropTypes.object).isRequired,
+
+    // options
+    direction: PropTypes.oneOf(['clockwise', 'counterclockwise']),
+    max: PropTypes.number,
+    min: PropTypes.number,
+    negativeColor: PropTypes.string,
+    negativeClass: PropTypes.string,
+    negativeLabel: PropTypes.string,
+    orientation: PropTypes.number,
+    size: PropTypes.number,
+    text: PropTypes.string
+  },
+
+  getDefaultProps () {
+    return {
+      data: [],
+      direction: 'clockwise',
+      orientation: 0,
+      min: 0,
+      negativeColor: '#ffffff',
+      negativeClass: 'negative-value',
+      negativeLabel: '',
+      size: 150,
+      _padding: 10
```

Comment:
I don't really care about this now - I think we've probably worked this one over enough, but the convention we're using for the proptypes and defaults is to group them by sections

// Required
data (required, so it doesn't need a default)
// Options
everything else
// Callbacks
// State
_padding